nav2_costmap_2d/src/costmap_2d.cpp

```diff
@@ -296,6 +296,21 @@ bool Costmap2D::worldToMap(double wx, double wy, unsigned int & mx, unsigned int
   return false;
 }
 
+bool Costmap2D::worldToMapContinuous(double wx, double wy, float & mx, float & my) const
+{
+  if (wx < origin_x_ || wy < origin_y_) {
+    return false;
+  }
+
+  mx = (wx - origin_x_) / resolution_ + 0.5;
```

Comment:
Cast the origin values to float before running this operation. I've been burned alot recently in MPPI/Smac with mixing types around and getting unexpected outcomes or massive performance hits

Reply:
`+ 0.5f`

Reply:
Wz, origin_x and resolution are double, I made the static cast on the result.